Add MHP Header and Footer

Blake-Haydon opened the pull request:
## Description

Enjoy website headers and footers? Well this PR is for you!
+ Footer look finalized
+ Header look finalized
+ Added styled components
+ Body content started (Still in development) 

## Screenshots
Header
![image](https://user-images.githubusercontent.com/23159604/89099371-05dc5300-d432-11ea-9382-8910682060e8.png)

Footer
![image](https://user-images.githubusercontent.com/23159604/89099385-1c82aa00-d432-11ea-8a22-d316350d205b.png)


## Steps to Test (Netlify)
[View here](https://deploy-preview-4--mhp-test.netlify.app/)

## Steps to Test (Locally)

1. Clone the git repo
2. Make sure node and npm is installed
3. Run `npm install -g gatsby-cli` to install Gatsby
4. Run `npm install` to install all dependancies
5. Run `gatsby develop`

hallgchris commented on src/components/info_block.js:
```diff
@@ -0,0 +1,65 @@
+        </div>
+
+        {/* Image component */}
+        <div className="col-md" style={{ height: 500, border: "red solid 2px" }}>
```
I don't really know how css works so maybe it'll be fine when we have actual images, but for now the borders seem to be too wide, not in line with the content above/below (as I assume would be intended). Will this be an issue when we have images?

Blake-Haydon replied:
These components are still a work in progress. The red border is just a placeholder for an image.

hallgchris commented on src/styles/global.css:
```diff
@@ -1,8 +1,42 @@
 :root {
     --MHP-green: #ccf381;
     --MHP-orange: #f2a252;
     --MHP-purple: #4831df;
```
Not from this PR, but nitpick: According to the image I found on the drive, this should be `#4831d4`

Blake-Haydon replied:
Haha thanks!

hallgchris commented on src/styles/global.css:
```diff
@@ -1,8 +1,42 @@
+
+/* Make hamburger menu pure white */
+.navbar-dark .navbar-toggler-icon {
+    background-image: url("data:image/svg+xml;charset=utf8,%3Csvg viewBox='0 0 32 32' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath stroke='rgba(255,255,255, 1)' stroke-width='2' stroke-linecap='round' stroke-miterlimit='10' d='M4 8h24M4 16h24M4 24h24'/%3E%3C/svg%3E") !important;
```
Can this be in its own svg file?

Blake-Haydon replied:
It's much easier in CSS to have it inline. I've tried to import into CSS directly and it is never easy.

hallgchris commented on src/components/button.js:
```diff
@@ -0,0 +1,40 @@
+
+import Link from "./link"
+
+const ButtonLink = styled(Link)`
```
Actually coming back to this, this seems to be the way styled components are. Happy with it if it's what's done lol, idk what I'm doing

Blake-Haydon replied:
Styled components significantly simplify adding/removing CSS. I tried using it in a global file and even with 6 components was getting confused.

hallgchris replied:
I don't know how it works (or which is better) but for comparison, in the dashboard repo, we seem to use this `.modules.css` thing for each component (e.g. https://github.com/monash-human-power/dashboard/blob/master/client/src/components/NavBar.module.css)

Blake-Haydon replied:
Yeah its basically CSS components except within the JS itself, instead of a separate `.module.css` file. This is significantly easier to debug as everything is confined to its class. 

If you add a style such as `H1 { font-size: 1000px }` in a .module.css it will pollute the global CSS namespace when imported.

khanguslee replied:
Yeah this is a styled component. I've played around with all the different CSS styles and I found that SASS was my go-to because:
- It mimics the CSS styling closely so moving from normal CSS to SCSS is really simple
- I don't like styling inside the JS files because I feel like it can lead to very large `.js` files and I don't really care about what the component looks like unless I want to adjust the styling (my major gripe about Vue as well).

Happy for you to use styled components since this was the decision you made though. How have you found it?

Blake-Haydon replied:
Compared to CSS modules it is much easier to deal with as the majority of components do not have much logic. If they had more logic I would use SASS in separate files.

khanguslee replied:
yeah thats fair enough. See how you go with it! I've been meaning to try out LESS as well just to play around with more CSS frameworks. (CSS is a real programming language boiz)

hallgchris replied:
Sounds good then, styled CSS!

hallgchris commented:
Blake the time traveler
![image](https://user-images.githubusercontent.com/17876556/89101510-d20a2900-d443-11ea-8db7-11b8388c8a74.png)

Blake-Haydon commented on src/components/button.js:
```diff
@@ -0,0 +1,40 @@
+`
+
+const Button = ({ children, href }) => (
+    <>
```
Nope, removed them!

nix-chu commented on src/components/footer.js:
```diff
@@ -0,0 +1,71 @@
+            <div className='row'>
+                {/* Col is xl as it should always collapse */}
+                <div className='col-xl pt-5'>
+                    © {new Date().getFullYear()}, Monash Human Power
```
Looks like the copyright symbol, I think it should be fine

hallgchris replied:
I know, I think we often try to steer clear of these symbols for compatibility reasons, replacing them with the codes found [here](https://www.w3schools.com/html/html_symbols.asp)

Blake-Haydon replied:
An easy fix, I'll replace it with the code

nix-chu commented on src/components/header.js:
```diff
@@ -1,42 +1,117 @@
 }
+`
+
+const Navbar = styled.nav`
+padding-right: 0px;
+padding-top: 0px;
+padding-bottom: 0px;
+`
+
+const ColapsingDiv = styled.div`
+  justify-content: space-between;
+`
```
No indentation vs 2 space indentation vs 4 space indentation is very triggering lmaoo

Blake-Haydon replied:
Just ran a formatter and it fixed it!

nix-chu commented on src/components/footer.js:
```diff
@@ -0,0 +1,71 @@
+                <div className='col-md'>
+                    <FooterHeading>Join Us</FooterHeading>
+                    <FooterParagraph>
+                        Connect with us on <FooterLink to='https://www.facebook.com/MonashHPT/'> Facebook</FooterLink>
```
Maybe you could move the social icons under here, since they are all related (unless Corporate says otherwise)

Blake-Haydon replied:
I'll have to ask corporate about that one. It does make sense to group them together.

nix-chu commented on src/components/button.js:
```diff
@@ -0,0 +1,40 @@
+import React from "react"
```
I've never done anything with React before, but are semicolons still required? Website works well locally though

Blake-Haydon replied:
Don't think so as it gets converted into _good_ js when gatsby builds the site.

khanguslee replied:
Yeah.. can you turn on semicolons? If you disable semicolons, it may lead to bugs because of javascript. See https://www.freecodecamp.org/news/codebyte-why-are-explicit-semicolons-important-in-javascript-49550bea0b82/

nix-chu replied:
A good read, never looked at semicolons that way before.

Blake-Haydon commented on src/components/button.js:
```diff
@@ -0,0 +1,40 @@
+)
+
+Button.propTypes = {
+    children: propTypes.node.isRequired,
```
Prettier is installed, just forgot to run it. It would be good if the formatter ran on each PR. 

Can this be done through GitHub actions?

khanguslee replied:
I always have Prettier run on save. If you want to force it on people, you can also use husky to run prettier and eslint everytime you commit!

Blake-Haydon replied:
> I always have Prettier run on save
I do as well but sometimes it gets confused with the styling.
Just added husky to do automatic formatting on every commit!

Blake-Haydon commented on src/components/header.js:
```diff
@@ -1,42 +1,117 @@
+`
+
+const NavLink = styled(Link)`
+/* Nav links should be white and then green when hovered over */
```
Fixed when I ran prettier!

Blake-Haydon commented on src/pages/404.js:
```diff
@@ -6,9 +6,35 @@ import SEO from "../components/seo"
+        <div className="col py-5" style={{ textAlign: "center" }}>
+          <h1>404: PAGE NOT FOUND 🚲</h1>
+          <p>You just hit a route that doesn&#39;t exist... the sadness. 🚲</p>
+        </div>
```
The 404 page is not finished.

khanguslee commented on src/components/button.js:
```diff
@@ -0,0 +1,40 @@
+};
+
+Button.defaultProps = {
+    children: 'Example Button Text',
```
Think I would prefer this to be empty string `""`. A button should be dumb and not know what to render as text.

khanguslee commented on src/components/footer.js:
```diff
@@ -0,0 +1,71 @@
+            <div className='row'>
+                {/* Col is xl as it should always collapse */}
+                <div className='col-xl pt-5'>
+                    © {new Date().getFullYear()}, Monash Human Power
```
Kind of a nit for me, can you move `new Date().getFullYear()` into it's own variable? It looks a lot neater when the return for this is only just html and variables. All 'logic' should not be inside here.

khanguslee commented on src/components/header.js:
```diff
@@ -1,42 +1,117 @@
+padding-bottom: 0px;
+`
+
+const ColapsingDiv = styled.div`
```
spelling: `CollapsingDiv`

khanguslee commented on src/components/header.js:
```diff
@@ -1,42 +1,117 @@
+          {/* Page Links */}
+          <div>
+            <ul className='navbar-nav ml-auto'>
+              {navItem('About', '#about')}
```
Can you have the navitems stored as an array of objects?:

```js
const pageLinks = [{"title": "About", "link": "#about"}, {"title": "Bike", "#bike"}, ...];
```

The render here would just be:
```js
{pageLinks.forEach(item => navItem(item.title, item.link)}
```

Allows us to add additional nav items later on by just modifying the above array!

khanguslee commented on src/components/info_block.js:
```diff
@@ -0,0 +1,65 @@
+        </div>
+
+        {/* Image component */}
+        <div className="col-md" style={{ height: 500, border: "red solid 2px" }}>
```
Perhaps not for this PR, but when you do DO more work on this area, can you put styles like this into it's own file? or use a style component?

Blake-Haydon replied:
It's still a WIP, so inline styles are used as they're easier to quickly change.

khanguslee commented on src/components/socials.js:
```diff
@@ -0,0 +1,47 @@
+const Socials = () => {
+    return (
+        <MhpSocialContainer>
+            <SocialLink to='https://www.facebook.com/MonashHPT/'>
```
Might want to do that refactor here as well. `const socialLink = [...]` etc. etc.

This set's us up later so that corporate is able to edit this using a CMS.

Blake-Haydon replied:
Because the `FontAwesomeIcon` is needed to be imported I think I will leave it for another PR as it will be more intuitive to integrate it together once the CMS is running.

khanguslee replied:
Sounds good to me!

Blake-Haydon commented:
Just some small changes made. Fixing the main issues with the PR. 
This PR is mainly for:
- `src/components/socials.js`
- `src/components/socials.js`

khanguslee commented on src/pages/404.js:
```diff
@@ -6,9 +6,35 @@ import SEO from "../components/seo"
-  </Layout>
+    <h1
+      className="pt-5 pb-1"
+      style={{ textAlign: "center", color: "white", background: "var(--MHP-purple)", fontWeight: "bold", fontSize: "60px" }}
```
Any chance we can get the styles not hardcoded? :(

khanguslee commented on src/pages/index.js:
```diff
@@ -2,14 +2,63 @@ import React from "react"
+    <div
+      className='container-fluid py-5'
+      style={{
+        // height: '100vh',
```
Remove commented out styles/code

Blake-Haydon commented:
Mostly small fixes, but a few new things have been added.
- `husky` was added for automatic formatting
- ReadMe has been updated to better reflect the project (ie. added how to run gatsby)
- Fixed the 404 page as can be seen [here](https://deploy-preview-4--mhp-test.netlify.app/404)
